### PR TITLE
DRAFT release notes 0.6.18

### PR DESCRIPTION
**Pull request by lsgunnlsgunn**
Ugly release notes for 0.6.18

### CODE REVIEW

File: modules/release-notes/pages/sdk-release-notes.adoc [hunk 1]
```diff
@@ -33,7 +33,29 @@ This {release} version of the software and programming language should not be us
...
 == Highlights of what's new in {release}
 
-The {release} release primarily includes user-visible new features, internal enhancements, and fixes and updates to the version of {proglang} and the {proglang} base library that are supported. 
+The {release} release primarily includes internal enhancements, and fixes and updates to {proglang}, the {proglang} base library, and Candid. 
+
+The most significant new features and fixes include the following:
+
+* A fix to `+dfx upgrade+` that prevents a `+main+` thread panicked when `+Option::unwrap()+` is called on a `+None+` value. 
+* A fix to the name of the language service command to restore its compatibility with IDEs. 
+* Candid has been updated to support more Rust built-in types and with support for reference types.
```
Comment (chenyan-dfinity) [inline modules/release-notes/pages/sdk-release-notes.adoc:42]:
```suggestion
```
This is not related to dfx

File: modules/release-notes/pages/sdk-release-notes.adoc [hunk 2]
```diff
@@ -33,7 +33,29 @@ This {release} version of the software and programming language should not be us
...
+* The {proglang} compiler supports specifying command-line arguments using `+--args <file>+` and `+--args0 <file>+`.
```
Comment (hansl) [inline modules/release-notes/pages/sdk-release-notes.adoc:43]: Can we prefix those next few items with `[compiler]` or `[proglang]`? "A change to option blocks and option checks..." is misleading that it might apply to DFX.

**Review by hansl — APPROVED** [review]
LGTM, I'd wait for @chenyan-dfinity comment to be resolved but will leave it to you two.

File: modules/release-notes/pages/sdk-release-notes.adoc (hunk 1 above)
Comment (lsgunnlsgunn) [reply modules/release-notes/pages/sdk-release-notes.adoc:47]: Do you really want it removed or just labeled as a Candid change? I think people are going to want some type of release notes or changelog for Candid updates eveentually.

Comment (chenyan-dfinity) [reply modules/release-notes/pages/sdk-release-notes.adoc:47]: Candid has its own release notes here: https://github.com/dfinity/candid/blob/master/Changelog.md, and it's a public repo, so people can already see that. It makes sense to include it here if the change affects dfx. In this case, it's only a Rust crate change, it may cause more confusion to include in the dfx release notes.

Eventually, I would expect to have a separate release notes for Motoko when the repo becomes public, so that we can decouple more complexities around release notes.

File: modules/release-notes/pages/sdk-release-notes.adoc [hunk 3]
```diff
@@ -33,7 +33,33 @@ This {release} version of the software and programming language should not be us
...
+=== DFX
+
+* A fix to `+dfx upgrade+` that prevents a `+main+` thread panic when `+Option::unwrap()+` is called on a `+None+` value. 
+* A fix to the name of the language service command to restore its compatibility with IDEs. 
+
+=== Motoko
+
+* The {proglang} compiler supports specifying command-line arguments using `+--args <file>+` and `+--args0 <file>+`.
++
+The new options enable the compiler to read newline and NUL terminated arguments from a specified file name.
```
Comment (chenyan-dfinity) [inline modules/release-notes/pages/sdk-release-notes.adoc:49]:
```suggestion
```
This is not in this release, but the next release. The changelog above the version numbers means it's in a future release.

**Review by chenyan-dfinity — APPROVED** [review]
I reworded the Motoko notes a bit. LGTM.
